app/src/main/kotlin/at/bitfire/davdroid/ui/composable/BasicTopAppBar.kt

```diff
@@ -0,0 +1,34 @@
+package at.bitfire.davdroid.ui.composable
+
+import android.app.Activity
+import android.content.Intent
+import androidx.annotation.StringRes
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import at.bitfire.davdroid.R
+import kotlin.reflect.KClass
+
+@Composable
+fun Activity.BasicTopAppBar(@StringRes titleStringRes: Int, parentActivity: KClass<*>) {
```

Comment:
To call `onSupportNavigateUp()` (see below), `BasicTopAppBar` could either be

1. an extension function of `AppCompatActivity`, or
2. a normal composable that gets the "onNavigateUp" action as an argument (which will usually just be `onSupportNavigateUp()`)

Variant 1 would require less (redundant) code, but can only be used within an `AppCompatActivity`.
Variant 2 would be more clean and can be used in any context, also outside of `AppCompatActivity`.

So I think we should have both:

1. a regular `BasicTopAppBar` that gets `onNavigateUp` as an argument and that can be used in any context, for instance from `TasksCard`. +
2. for convenience, there can be a `fun AppCompatActivity.BasicTopAppBar` that calls the `BasicTopAppBar` and passes `onNavigateUp = onSupportNavigateUp`.